pkg/routeagent_driver/handlers/ovn/gateway_dataplane.go

```diff
@@ -155,22 +155,16 @@ func (ovn *Handler) setupForwardingIptables() error {
 }
 
 func (ovn *Handler) updateNoMasqueradeIPTables() error {
-	rules := ovn.getNoMasqueradRuleSpecs()
-
-	return errors.Wrapf(submiptables.UpdateChainRules(ovn.ipt, "nat", constants.SmPostRoutingChain, rules),
-		"error updating %q rules", constants.SmPostRoutingChain)
-}
-
-func (ovn *Handler) getNoMasqueradRuleSpecs() [][]string {
-	var rules [][]string
-
 	for _, endpoint := range ovn.remoteEndpoints {
 		for _, subnet := range endpoint.Spec.Subnets {
-			rules = append(rules, []string{"-d", subnet, "-j", "ACCEPT"})
+			err := ovn.ipt.AppendUnique("nat", constants.SmPostRoutingChain, []string{"-d", subnet, "-j", "ACCEPT"}...)
```

Comment:
It looks like we are not deleting the iptable rules when the remote endpoint is deleted.